packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx

```diff
@@ -2,16 +2,98 @@ import React, { HTMLAttributes } from "react"
 import classnames from "classnames"
 import { VisuallyHidden } from "~components/VisuallyHidden"
 import { OverrideClassName } from "~components/types/OverrideClassName"
-import styles from "./LoadingSpinner.module.scss"
+import styles from "./LoadingSpinner.module.css"
+
+type LoadingSpinnerSize = "xs" | "sm" | "md"
 
 export type LoadingSpinnerProps = {
   accessibilityLabel: string
   /**
-   * Generally use "md" unless spinner is inside a form field
+   * Generally use "md" unless spinner is inside a form field.  @default "md"
    */
-  size?: "sm" | "md"
+  size?: LoadingSpinnerSize
 } & OverrideClassName<Omit<HTMLAttributes<HTMLDivElement>, "children">>
 
+const renderLoadingSpinner = (size: LoadingSpinnerSize): JSX.Element => {
```

Comment:
I'd convert these to individual components instead of a function that returns a JSX.Element

Reply:
Sure could break them into small subcomponents for cleanliness. In terms of rendering them out, I suppose I could do a ternary (though two conditions are a little ick to me), otherwise just a function that have a switch? i.e:

```
    {() => {
      switch (size) {
        case "sm":
          return <SmallSpinner />
        case "xs":
          return <ExtraSmallSpinner />
        case "md":
          return <MediumSpinner />
        default:
          return null
      }
    }}
    {size === "sm" ? (
      <SmallSpinner />
    ) : size === "xs" ? (
      <ExtraSmallSpinner />
    ) : (
      <MediumSpinner />
    )}
```

Reply:
I suppose the ternary isn't too bad, a less code needed and we likely won't need any more than 3 sizes (likely)

Reply:
Or something like this :)

```suggestion
const Spinner = (size: LoadingSpinnerSize): JSX.Element => {
  if (size === "xs") return <ExtraSmallSpinner />
  if (size === "sm") return <SmallSpinner />
  return <MediumSpinner />
}
```

Reply:
Right, I getcha 😉

Reply:
I've moved all the spinners into a `SpinnerIcon` subcomponent, it was starting to look a little bloated and messy in the top of the component

Reply:
As long as you don't export it at the entrypoint then it should be fine :) Because currently `SpinnerIcon` exists so it would conflict 😬